### ISSUE TITLE
"invalid reference" error with `lifecycle.ignore_changes`

### ISSUE THREAD

**Issue by LiuVII** [terraform-linters/tflint#1474]
## Introduction

When `lifecycle` `ignore_changes` block contains `data` e.g.

```hcl
resource "kubernetes_secret" "my_secret" {
  metadata {
   ...
  }

  data = {
   ...
  }

  lifecycle {
    ignore_changes = [
      data
    ]
  }
}
```

Tflint shows a warning on it.

## Expected Behavior

This shouldn't cause any issues

## Actual behavior

Tflint shows this warning on this code
```
Error: Invalid reference; The "data" object must be followed by two attribute names: the data source type and the resource name.
```

## Additional Context

```console
$ tflint -v
TFLint version 0.39.2
$ terraform -v
Terraform v1.1.4
```

**Comment by wata727** [comment]
Thank you for reporting this.

This is probably caused by creating references for all expressions including `ignore_changes` since v0.39.0. See https://github.com/terraform-linters/tflint/pull/1432.
Normally, unknown references are ignored, but unfortunately the `data` seems to be misidentified as a reference to a data resource here:
https://github.com/terraform-linters/tflint/blob/v0.39.2/rules/terraformrules/utils.go#L291-L299

This issue doesn't occur in Terraform as it doesn't generate references to `ignore_changes`. We should avoid this case in its own way. It's a hard question to work around, but running `terraform validate` can catch reference errors, so one idea is to ignore the errors for now.

**Comment by LiuVII** [comment]
Thank you!
